### DISCUSSION TITLE
alpine does not get updated after livewire request

### DISCUSSION THREAD

**Discussion by rgroli** [alpinejs/alpine#1748] [7. Bugs (archived)] [answered]
Hi, I have strange problem, with alpine and livewire playing together...

Livewire: 
```php
<?php

namespace Cms\Http\Livewire;

use Cms\Http\Livewire\Component;

class Test extends Component
{

    public $value;

    public function mount()
    {
        $this->value = time();
    }

    public function render()
    {
        return view('livewire.test');
    }

    public function refreshValue()
    {
        $this->value = time();
    }
}
```

Blade / Alpine
```blade
<div>
    <!-- Current Timestamp in millseconds / This gets updated on every refresh -->
    {{ $value }}

    <!-- Getting a new value -->
    <button wire:click="refreshValue">click</button>

    <div x-data="{value: {{ json_encode($value) }}}">
        <!-- AFTER Livewire roundtrip this does not get updated, although the value in x-data changed -->
        <span x-text="value"></span>
    </div>
</div>
```

Do I need to do something else to make alpine recognize the changes in x-data ?

**Comment by anatoliy-t7** [comment]
I have a same issue.

**Comment by inmanturbo** [comment]
This must be the same issue I'm having here: https://github.com/alpinejs/alpine/discussions/1738

As in my case x-data is getting updated, however x-show doesn't seem to be aware of the change.

**Reply by rgroli** [reply]
Yes, that seems related.

**Comment by joshhanley** [comment, marked as answer]
`x-data` won't get re-evalulated once it has already loaded, so it won't pick up the changes (even if you can see them in the outputted HTML).

To keep data in sync you should use `@entangle` or query the Livewire component directly using `$wire`.

So for example

```blade
<div>
    <!-- Current Timestamp in millseconds / This gets updated on every refresh -->
    {{ $value }}

    <!-- Getting a new value -->
    <button wire:click="refreshValue">click</button>

    <div x-data="{ value: @entangle('value') }">
        <span x-text="value"></span>
    </div>
</div>
```

See "Sharing State Between Livewire And Alpine" here in the docs https://laravel-livewire.com/docs/2.x/alpine-js#:~:text=sharing%20state%20between%20livewire%20and%20alpine%3A%20%40entangle

Hope this helps!

**Reply by rgroli** [reply]
@joshhanley That did the trick, thanks!

**Reply by joshhanley** [reply]
No worries 🙂
